File: app/type/model/task.go
package model

import "time"

type Task struct {
	ID          uint64     `gorm:"primaryKey;AUTO_INCREMENT;column:id;" json:"id" example:"1"` // ID
	ProcessID   uint64     `gorm:"column:process_id;default:0" json:"process_id" example:"1"`
	ProcessCode string     `gorm:"column:process_code;default:''" json:"process_code" example:"1"`
	ProcessName string     `gorm:"column:process_name;default:''" json:"process_name" example:"1"`
	ExecutionID uint64     `gorm:"column:execution_id;default:0" json:"execution_id" example:"1"`
	NodeID      uint64     `gorm:"column:node_id;default:0" json:"node_id" example:"1"`
	NodeCode    string     `gorm:"column:node_code;default:''" json:"node_code" example:"1"`
	NodeName    string     `gorm:"column:node_name;default:''" json:"node_name" example:"1"`
	AssigneeID  string     `gorm:"column:assignee_id;default:''" json:"assignee_id" example:"1"`
	Status      string     `gorm:"column:status;default:'running'" json:"status" example:"running"`
	StartedAt   *time.Time `gorm:"column:started_at" json:"started_at"`
	EndedAt     *time.Time `gorm:"column:ended_at" json:"ended_at"`
	Remark      string     `gorm:"column:remark;default:''" json:"remark" example:"1"`
	CreatedAt   time.Time  `gorm:"column:created_at;autoCreateTime;" json:"created_at" example:"2020-10-11T10:10:10"` // 创建时间
	UpdatedAt   time.Time  `gorm:"column:updated_at;autoUpdateTime;" json:"updated_at" example:"2020-10-11T10:10:10"` // 更新时间
}

func (*Task) TableName() string {
	return "task"
}

File: web/app/types/task.ts
export interface Task {
  id: number
  process_id: number
  process_code: string
  process_name: string
  execution_id: number
  node_id: number
  node_code: string
  node_name: string
  assignee_id: number
  assignee_name?: string
  status: 'pending' | 'running' | 'completed' | 'skipped'
  started_at: string | null
  claimed_at: string | null
  ended_at: string | null
  remark: string
  created_at: string
  updated_at: string
}

export interface TaskQueryBody {
  page: number
  size: number
  process_id?: number
  process_code?: string
  execution_id?: number
  node_id?: number
  node_code?: string
  assignee_id?: number
  status?: string
}

export interface TaskClaimableQueryBody {
  page: number
  size: number
  status?: string
}

export interface TaskQueryData {
  total: number
  items: Task[]
}

export interface TaskClaimBody {
  id: number
}

export interface TaskCompleteBody {
  id: number
  assignee_id: number
  remark?: string
}

export interface TaskSkipBody {
  id: number
  assignee_id: number
}

export interface TaskCandidate {
  id: number
  task_id: number
  user_id: number
  source_type: 'user' | 'role'
  source_id: number
  user_name?: string
  created_at: string
  updated_at: string
}

export interface TaskCandidateListBody {
  task_id: number
}

export type TaskCandidateListData = TaskCandidate[]

export interface TaskLog {
  id: number
  process_id: number
  process_code: string
  execution_id: number
  node_id: number
  node_code: string
  task_id: number
  action: string
  assignee_id: string
  assignee_name?: string
  remark: string
  created_at: string
}

export interface TaskLogQueryBody {
  page: number
  size: number
  task_id?: number
  execution_id?: number
}

export interface TaskLogQueryData {
  total: number
  items: TaskLog[]
}
